### PR TITLE
Fixing make stub warnings and mypy issues

### PR DESCRIPTION
**Pull request by sw23**
This cleanup is part of #2041 - cleaning up warnings from `make stubs` and `make check-stubs`.  Adding more detailed type information and fixing a few types.

I worked on this as part of #hacktoberfest (https://hacktoberfest.digitalocean.com/), so if the PR looks OK would you mind adding the label `hacktoberfest-accepted`?

Note that several of the warnings/errors are coming from the ulab package, so I'll try and make a PR there as well.
[Update] Looks like most of the ulab errors have been fixed now

### CODE REVIEW

File: shared-bindings/displayio/Display.c [hunk 1]
```diff
@@ -39,7 +39,7 @@
 #include "shared-module/displayio/__init__.h"
 #include "supervisor/shared/translate.h"
 
-//| _DisplayBus = Union[FourWire, ParallelBus, I2CDisplay]
+//| _DisplayBus = Union['FourWire', 'ParallelBus', 'I2CDisplay']
```
Comment (tannewt) [inline shared-bindings/displayio/Display.c:42]: Why are these needed? I'm new to typing.

File: shared-bindings/wifi/Radio.c [hunk 2]
```diff
@@ -283,7 +283,7 @@ const mp_obj_property_t wifi_radio_ap_info_obj = {
                (mp_obj_t)&mp_const_none_obj },
 };
 
-//|     def ping(self, ip, *, timeout: float = 0.5) -> float:
+//|     def ping(self, ip: wifi.Radio, *, timeout: float = 0.5) -> float:
```
Comment (tannewt) [inline shared-bindings/wifi/Radio.c:286]: I think this should be an `ipaddress.IPv4Address`.

**Review by tannewt — CHANGES_REQUESTED** [review]
One question and type correction. Thank you!

File: shared-bindings/displayio/Display.c (hunk 1 above)
Comment (sw23) [reply shared-bindings/displayio/Display.c:42]: Quotes around custom types are helpful when the type definitions haven't yet been defined (in this case because the type definitions come later in the same file).  This doesn't bother mypy but pylint complains about this.

Pylint error: "Using variable 'FourWire' before assignment"
...
More Info: https://www.python.org/dev/peps/pep-0484/#forward-references

File: shared-bindings/wifi/Radio.c (hunk 2 above)
Comment (sw23) [reply shared-bindings/wifi/Radio.c]: Good catch! Fixed up now

**Review by tannewt — APPROVED** [review]
Thank you! Is there a mypy check you could add to our CI in a follow up PR?

**Comment by sw23** [comment]
@tannewt - Yep, will do!
